### DISCUSSION TITLE
Blocking a specific message from a device

### DISCUSSION THREAD

**Discussion by Littleislandbrewing** [Koenkk/zigbee2mqtt#17330] [General]
hello peeps.

I have a bunch of Tuya Radars and they are spamming the network. 

I have managed to slow down the amount of spam by using debounce: 1   and it works reasonably well with proper messages - however there is still some noisy spam that looks like this>

```
2023-04-14 09:38:16Received Zigbee message from 'entrance radar', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[0],"type":"Buffer"},"datatype":4,"dp":1}],"seq":56832}' from endpoint 1 with groupID 0
Debug 2023-04-14 09:38:16Received Zigbee message from 'entrance radar', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[0,0,3,182],"type":"Buffer"},"datatype":2,"dp":4}],"seq":57088}' from endpoint 1 with groupID 0
Debug 2023-04-14 09:38:16Received Zigbee message from 'entrance radar', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[0,0,0,9],"type":"Buffer"},"datatype":2,"dp":2}],"seq":57344}' from endpoint 1 with groupID 0
Debug 2023-04-14 09:38:16Received Zigbee message from 'entrance radar', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[0,0,0,60],"type":"Buffer"},"datatype":2,"dp":3}],"seq":57600}' from endpoint 1 with groupID 0
Debug 2023-04-14 09:38:16Received Zigbee message from 'entrance radar', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[1],"type":"Buffer"},"datatype":4,"dp":6}],"seq":57856}' from endpoint 1 with groupID 0
Debug 2023-04-14 09:38:16Received Zigbee message from 'entrance radar', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[0,0,0,0],"type":"Buffer"},"datatype":2,"dp":101}],"seq":58112}' from endpoint 1 with groupID 0
Debug 2023-04-14 09:38:16Received Zigbee message from 'entrance radar', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[0,0,2,88],"type":"Buffer"},"datatype":2,"dp":102}],"seq":58368}' from endpoint 1 with groupID 0
Debug 2023-04-14 09:38:16Received Zigbee message from 'entrance radar', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[0,0,0,0],"type":"Buffer"},"datatype":2,"dp":9}],"seq":58624}' from endpoint 1 with groupID 0
Debug 2023-04-14 09:38:16Received Zigbee message from 'entrance radar', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[32],"type":"Buffer"},"datatype":3,"dp":103}],"seq":58880}' from endpoint 1 with groupID 0
Debug 2023-04-14 09:38:17Received Zigbee message from 'entrance radar', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[0,0,0,17],"type":"Buffer"},"datatype":2,"dp":104}],"seq":59136}' from endpoint 1 with groupID 0`

```

How do I filter on type 'commandDataReport'  or 'dpValues'    for the whole message???

**Comment by Koenkk** [comment]
AFAIK there is no way to reduce the spamming of these TuYa devices.

**Reply by tothgergely92** [reply]
And what about other Tuya device, namely ZY-M100 human presence sensors? Do you have any info to lower the messages they emit? Not only the mqtt network flooded, but also Home Assistant DB is becoming quite big because of these "3-4 in a second" state reports :(

**Reply by pawel-tomkiel** [reply]
For ZY-M100 please take a look at this comment: https://github.com/Koenkk/zigbee2mqtt/issues/19045#issuecomment-2362489362

**Comment by aatifk** [comment]
@Littleislandbrewing what do you mean by debounce? I’m having the same issue and I feel anything will help. Can you please explain what you did and how I can do the same?

@Koenkk it’d be amazing if you can figure out a way to fix this, honestly. The network is getting bogged down now.

**Comment by Koenkk** [comment]
Sorry, but I don't know a way to reduce TuYa message spamming, I believe it also occurs when using it with their gateway.

**Reply by aatifk** [reply]
Thank you for replying. Would it affect the network performance? I haven’t noticed an impact yet, so should I just ignore it or consider switching to the Wi-Fi version of this sensor?

**Reply by kkossev** [reply]
> Sorry, but I don't know a way to reduce TuYa message spamming, I believe it also occurs when using it with their gateway.

Yes, the same spamming occurs when Tuya mmWave sensors are paired to Tuya Zigbee gateway.

What helped (kind of) in another home automation platform was filtering the Distance reports in a very early stage of processing the Zigbee messages, so that there is no any kind of logging or further processing.

**Reply by Koenkk** [reply]
@aatifk you shouldn't have too many in your network, a few shouldn't be a problem.

**Reply by aatifk** [reply]
@Koenkk thanks again for the response. I have four while using a SONOFF Dongle P, and not planning on adding any more. Hoping this won't cause issues down the line, but we shall see.

**Comment by bjornarfjelldal** [comment]
I had 6 of these on my Conbee, and it killed the whole network. Have since then removed all and switched to the SkyConnect - wondering if I should try adding one or two again now.. Does debounce then also delay the time from sensing to logging in HA?

**Reply by aatifk** [reply]
I reduced my four to just one, substituting others to WiFi based ones (Aqara FP2 mainly). The one I have now hasn't caused any issues yet. Debounce didn't really help - it kind of killed the presence logging in HA though I probably gave up before trying to tinker with the debounce settings. 

FWIW, when I was running for, it was on SONOFF Zigbee Dongle P and while the network was being spammed, it didn't really bring things down. My understanding is that the SONOFF is a better coordinator than Conbee anyway.

**Comment by thelogman** [comment]
This is going to become more and more of an issue as these devices gain in popularity and fall in price. 

To me this is a perfect example of why sometimes 2 Zigbee networks makes sense. The chatty devices could be placed on a separate coordinator using a different topic than the main network and then let them do all the talking they want amongst themselves. 

It would be really nice if Z2M had a built in accommodation to handle multiple coordinators running on the same MQTT broker while using separate topics. 

--Barry

**Reply by diagonali** [reply]
That would make sense, to have chatty devices on a separate network, but wouldn't there still be the issue of the mqtt server still being overloaded with messages? I've just finally disconnected a ridiculously chatty tuya mmwave sensor from zigbee2mqtt for the reasons discussed in this thread and unfortunately it seems the only option is to move over to an Aqara device. Other than spamming messages the tuya mmwave devices are excellent.

**Reply by davidohne** [reply]
Just an idea since I recently moved from FHEM (another home automatisation software) to HASS and I am running both of them at the same time at the moment. Some of my "old" devices aren't implemented in home assistant and will never be. So I connected FHEM via MQTT to homeassistant. FHEM subscribes to commands TO the old devices and publishes messages from them to HASS. 

Perhaps we could use that kind of "system" to dump the flooding messages of the radar sensors: You run another zigbee2mqtt instance with another dongle to open a complete new network and connect all the radar sensors to that instance. You connect this zigbee2mqtt instance to your second mqtt broker (e.g. FHEM or another HASS instance) and redirect only the wanted messages to your main broker. There will be a little delay but I don't think you would kind of feel it in real life. That way, you would be able to protect your main mqtt broker and your main zigbee network, but well. It's a lot of effort, and using the Wi-Fi version (which hopefully doesn't bring down your 2.4ghz network) can be the easier way. But if you are running Proxmox/ESXI anyway (so it's easy to create a new LXC or VM) and have some hardware around, it could be worth a try.

**Reply by diagonali** [reply]
I think that would work, to isolate the mqtt server by creating..... another mqtt server! It means the spammy mmwave device is getting it's own way and effectively requires a whole zigbee2mqtt and mqtt installation all to itself. It's an intriguing "solution" and the nerd in me is curious to give it a try but I don't think (sadly) that it's practical and I'm not sure I've got the time to get it going any time soon unfortunately. I did try the local tuya HACS integration and connected the chatty tuya mmwave sensor to an official (and very cheap) tuya gateway I bought a while ago for this very purpose. I quickly found out that the "local tuya" HACS integration doesn't work with zigbee devices for some reason and neither could I get it to work with the gateway itself. Ah well...

**Reply by davidohne** [reply]
Hm that's unfortunate. I ordered my devices yesterday, but I didn't see the problem until today, otherweise I would have canceled the order. My devices will take 2 weeks to arrive from china and I will then try my luck with the workaround. It feels like creating new VMs is part of my daily routine, so it won't be a big thing. Otherweise it still sounds and fills like a dirty solution. Better filtering would be the cleaner way, but therefore we would need to change the device's firmware I think.

**Reply by diagonali** [reply]
Yeah I think modifying the firmware would be the only "real" solution and would turn these into super cheap and otherwise excellent sensors but that's way beyond my current capabilities. Hope all goes well with the workaround. As an aside, I did run one of these devices connected directly to zigbee2mqtt for a couple of years and used a "debounce" value of 1-2 which seemed to work quite well, probably fine even. It won't address (I think!) zigbee2mqtt being flooded in the background but in the logs it calms things down a lot. I had to do a lot of trial and error with the debounce value and the settings on the device itself ("fading time" etc which btw in the official app can't go below 10s) and while results were "ok", there were some remaining inconsistencies in detection and clearing detection. It wasn't quite stable enough for my use (I had some recent zigbee sensor lag/delay throughout the network which actually turned out to be wifi interference but which prompted me to rethink the tuya mmwave sensor) but it might work for your use case so possibly worth trying before spinning up a VM.

**Reply by jvdburgt** [reply]
A bit late to the party, but I put 8 of these sensors on a separate zigbee network, and all seems to work fine now. It's a bit of a crude workaround to a problem that should be solved at the device level, but at least it works.

**Comment by davidohne** [comment]
So the tuya zigbee presence detector arrived two days ago. When you relied on infrared motion detector the past years and all your family members hated you because of the false-non-detection... this seems to be a new chaptor now. No more lights off when people are sitting still in the room 🗡️ 

Anyway: Yes they are spammy and like I said I excluded them from my main network by creating a new VM on my ESXI host. I Only had an old CC2531, but well. It's working.

What I did and if someone is stumbling here:
1. Install zigbee2mqtt on a new VM
2. Change the channel of the new zigbee2mqtt instance to avoid interferences with your main network
3. I used FHEM as "filtering"-Smarthome system. Here a MQTT server is integrated 
4. Connect zigbee2mqtt to FHEM and add the radar device to the zigbee network: (German) https://wiki.fhem.de/wiki/Zigbee2mqtt
5. Connect the FHEM-Instance via MQTT to your Home Assistant Instance: (German) https://forum.fhem.de/index.php?topic=115279.0


I wanted to exclude all the spammy messages from my main zigbee network and my homeassistant instance. Perhaps you could just use a second zigbee network and directly connect to the main homeassistant instance. That would avoid lots of configuration on the second smarthome software, where I filter everything and just send the presence payload to the home assistant mqtt broker. 

It works. Perhaps it would've also worked by just adding the radar devices to my main network and don't give a F... Currently I don't know how I should go ahead: Take the wifi versions since wifi is more used to so much spammy messages and device interaction or stick to the plan and have a messy zigbee network with all radar devices...

**Reply by markhaines** [reply]
I've installed 4 of these Tuya mmWave devices and have another 7 I plan to install. I bought the ceiling install versions but am now having second thoughts as I don't want to kill my network.  Think I might try the second z2m instance and try that. Are you still happy with how it's performing a couple of months later?

If I replace them with the wifi version I really don't want to have to be reliant on Tuya, I've only just got rid of all my tuya sockets...

**Reply by jvdburgt** [reply]
Fwiw, I now have 12 sensors on a dedicated 2nd network (z2m), and it seems to be working correctly so far (3 days and counting).

**Reply by markhaines** [reply]
cool. i think i'm going to buy one of the new sonoff E spec dongles and then spin up a separate z2m instance in docker and swing them over. on the home assistant side is it just a case of pointing them both at the same MQTT Mosquito instance but give them different topic names?
